product_set/models/product_set.py

```diff
@@ -32,14 +32,20 @@ class ProductSet(models.Model):
         "it's going to be available for all of them.",
     )
 
+    display_name = fields.Char(
+        compute="_compute_display_name", string="Display Name", store=True
+    )
+
     def name_get(self):
-        return [(rec.id, rec._name_get()) for rec in self]
+        return [(rec.id, rec.display_name) for rec in self]
```

Comment:
If you implement a display_name, you don't need to override name_get() (moreover deprecated in 17.0).

See : https://github.com/odoo/odoo/blob/17.0/odoo/models.py#L1676